modules/api/src/main/java/net/countercraft/movecraft/events/CraftTranslateEvent.java

```diff
@@ -17,14 +19,16 @@
     private static final HandlerList HANDLERS = new HandlerList();
     @NotNull private final BitmapHitBox oldHitBox;
     @NotNull private final BitmapHitBox newHitBox;
+    @NotNull private final World world;
     @NotNull private String failMessage = "";
     private boolean playingFailSound = true;
     private boolean isCancelled = false;
 
-    public CraftTranslateEvent(@NotNull Craft craft, @NotNull BitmapHitBox oldHitBox, @NotNull BitmapHitBox newHitBox) {
+    public CraftTranslateEvent(@NotNull Craft craft, @NotNull BitmapHitBox oldHitBox, @NotNull BitmapHitBox newHitBox, @NotNull World world) {
```

Comment:
Same as previous comment, if the field isn't modifiable and can be reached from elsewhere, don't give it a constructor parameter nor a getter.

Reply:
The craft's world is not changed until after this event is called

Reply:
Ah. Is there anything stopping us from changing the world beforehand?

Reply:
Yes, both worlds are used later in TranslateTask so changing the world in the middle would require an extra variable. Also including both the world of origin and destination world might be useful to developers using the event.

Reply:
Technically it is possible but I think it would just end up making the following references to the worlds more confusing.

Reply:
Yep you're right